src/utils.ts

```diff
@@ -0,0 +1,26 @@
+import * as glob from 'glob';
+
+/**
+ * Checks for violations that might exist within the provided directory path.
+ * Keep in mind that due to malfunction of the [bsorrentino/maven-confluence-plugin](https://github.com/bsorrentino/maven-confluence-plugin) the naming of the pages can't be done via spaces. For now we agreed to replace the spaces with underscores.
+ * @param rootPath Path of the directory that needs to be checked agains the naming convention.
+ */
+export function checkForNamingViolations(rootPath: string) {
+  const WHITESPACE_REGEX = /\s/;
+  const directoryWithViolations = glob
```

Comment:
I have omitted those of recursive and array flattening functions and replaced it with a `glob`. It's  a very popular npm package and it's nice for doing such stuff in a node environment and makes the code much cleaner.